lib/stdlog.js

```diff
@@ -70,17 +70,54 @@
 
     // Use tail -F logs/generator_latest.txt to see the latest output
 
-    function writeToLog(data, encoding) {
+    function writeToLog(chunk, encoding, colorFunction) {
+        // We stopped logging in a previous call to write: return now
+        if (_logBytesWritten > LOG_SIZE_LIMIT) { return; }
+
+        var chunkLength,
+            logMsg;
+
+        if (!Buffer.isBuffer(chunk)) {
+            chunk = String(chunk);
+            if (colorFunction) {
+                chunk = colorFunction(chunk);
+            }
+            chunkLength = Buffer.byteLength(chunk, encoding);
+        } else {
+            chunkLength = chunk.length;
+        }
+
+        // Calculate the new log file size
+        _logBytesWritten += chunkLength;
+
+        // Limit not yet reached: write to log file
+        if (_logBytesWritten > LOG_SIZE_LIMIT) {
+            logMsg = "The log file size limit of " + LOG_SIZE_LIMIT + " bytes was reached." +
+                " No further output will be written to this file.";
+        } else {
+            logMsg = chunk;
+        }
+
         _logFilePaths.forEach(function (path) {
             try {
-                fs.appendFileSync(path, data, { encoding: encoding });
+                fs.appendFileSync(path, logMsg, { encoding: encoding });
             } catch (e) {
                 // do nothing
             }
         });
     }
 
-    function logStream(stream, colorFunction) {
+    function logReadableStream(stream) {
```

Comment:
Readable and Writable are two different types of [streams](https://nodejs.org/dist/latest-v4.x/docs/api/stream.html#stream_stream).  The previous version `logStream` tapped into a writable stream (stdout).  I renamed it to `logWritableStream` to differentiate the newer `logReadableStream` which listens to `data` events from a readable stream (generator logStream).  I'm open to suggestions on clearer names.  

Also I'll welcome a second opinion on this PR by @iwehrman, I forgot to tag him until now.  Word on the street is that he has a stream fetish.